### ISSUE TITLE
Problem(?) with switching between backend tracers

### ISSUE THREAD

**Issue by nutchalum** [DataDog/dd-trace-js#81]
I tried to migrate from `jaeger` to `ddtracer`

both instances are created from `opentracing`

```javascript
 console.log(process.env.TRACER)

 const span = tracer.startSpan(host, { childOf: parentContext })

//this is where I try to debug
 console.log(span.context())
```

result when using `jaeger`
```
debug: jaeger
debug:  
{ _traceId: 
   { '0': 99,
     '1': 109,
     '2': 24,
     '3': 134,
     '4': 131,
     '5': 164,
     '6': 174,
     '7': 156 },
  _spanId: { '$ref': '$["_traceId"]' },
  _parentId: null,
  _traceIdStr: null,
  _spanIdStr: null,
  _parentIdStr: null,
  _flags: 0,
  _baggage: {},
  _debugId: '',
  _samplingFinalized: false }
```

result when using `ddtracer`
```
debug: datadog
debug:  
{ traceId: { buffer: [Object], offset: 0 },
  spanId: { '$ref': '$["traceId"]' },
  parentId: null,
  sampled: true,
  baggageItems: {},
  trace: { started: [Array], finished: [] } }
```

It seems like the contexts are entirely different. So, what do I probably miss?

**Comment by rochdev** [comment]
The span context is actually an implementation specific bag. Its main use is to propagate context both internally and across services. In general, it should not be used directly but in conjunction with `tracer.inject()` and `tracer.extract()`.

See https://doc.esdoc.org/github.com/opentracing/opentracing-javascript/class/src/span_context.js~SpanContext.html

What is your use case?

**Comment by nutchalum** [comment]
My code is kind of complicated, maybe it's not about the context at all.

Before this, I use jaeger and it works normally.
The problem happens when I try to use `ddtracer` and make a http request (axios as a client).
So I inspect the context to see what's the difference? (because it used to work)

This is the error I got and this message hit me `Cannot destructure property 'span' of 'undefined' or 'null'` because it is not both of that (null or undefined).

```
error:  TypeError: Cannot read property 'href' of undefined
    at Object.requestTrace [as request] (/myproject/node_modules/dd-trace/src/plugins/http.js:19:34)
    at RedirectableRequest._performRequest (/myproject/node_modules/follow-redirects/index.js:128:24)
    at new RedirectableRequest (/myproject/node_modules/follow-redirects/index.js:54:8)
    at Object.wrappedProtocol.request (/myproject/node_modules/follow-redirects/index.js:252:14)
    at dispatchHttpRequest (/myproject/node_modules/axios/lib/adapters/http.js:132:25)
    at new WrappedPromise (/myproject/node_modules/async-listener/es6-wrapped-promise.js:13:18)
    at httpAdapter (/myproject/node_modules/axios/lib/adapters/http.js:18:10)
    at dispatchRequest (/myproject/node_modules/axios/lib/core/dispatchRequest.js:59:10)
    at propagateAslWrapper (/myproject/node_modules/async-listener/index.js:502:23)
    at /myproject/node_modules/async-listener/glue.js:188:31
debug:  TypeError: Cannot destructure property `span` of 'undefined' or 'null'.
    at afterRequestErrorHandler (/myproject/src/http-clients/axios.js:110:26)
    at propagateAslWrapper (/myproject/node_modules/async-listener/index.js:502:23)
    at /myproject/node_modules/async-listener/glue.js:188:31
    at proxyWrapper (/myproject/node_modules/async-listener/index.js:511:29)
    at proxyWrapper (/myproject/node_modules/async-listener/index.js:511:29)
    at /myproject/node_modules/async-listener/index.js:539:70
    at /myproject/node_modules/async-listener/glue.js:188:31
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:182:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:697:11)
```

To be honest, I have no idea what the problem really is.

**Comment by rochdev** [comment]
It seems to be a problem with the automatic context propagation that is done internally. This is not available when using OpenTracing, but it seems it's still trying to do it. Definitely a bug.

**Comment by rochdev** [comment]
@nutchalum Is there any place where this `span` property is accessed in your code?

**Comment by nutchalum** [comment]
Yes, I'm using axios interceptors https://github.com/axios/axios#interceptors

I have to assign `span` along with `config` so I can use it in response

```javascript
// Add a request interceptor
axios.interceptors.request.use(function (config) {
    // Do something before request is sent
    config.span = span

    return config;
  }, function (error) {
    // Do something with request error
    return Promise.reject(error);
  });

// Add a response interceptor
axios.interceptors.response.use(function (response) {
    const { span } = response.config
    // Do something with response data

    return response;
  }, function (error) {
    // Do something with response error
    return Promise.reject(error);
  });
```

**Comment by rochdev** [comment]
The first error will be fixed in the next release today.

The second error is because the `response` object that is returned by axios doesn't seem to have a `config` property at this line: `const { span } = response.config`.

**Comment by rochdev** [comment]
Should be fixed in `v0.1.6`

@nutchalum Can you confirm?

**Comment by nutchalum** [comment]
Fixed, everything is green now. 
Thank you @rochdev
